### PR TITLE
Fix dependency resolution logic

### PR DESCRIPTION
**Pull request by jmarolf**
Allows other repos to opt into getting binding redirects updated on insertion.

### CODE REVIEW

File: src/RoslynInsertionTool/RoslynInsertionTool/RoslynInsertionTool.Packages.cs [hunk 1]
```diff
@@ -89,12 +89,12 @@ public OutdatedPackageException(string message, PackageInfo package, SemanticVer
...
-                var buildVersion = package.Version.GetSuffixBuildVersion();
+                var packageBuildVersion = package.Version.GetSuffixBuildVersion();
 
-                if (buildVersion.Build != roslynBuildVersion.FiveDigitBuildNumber ||
-                    buildVersion.Revision != roslynBuildVersion.Revision)
+                if (packageBuildVersion.Build != buildVersion.FiveDigitBuildNumber ||
+                    packageBuildVersion.Revision != buildVersion.Revision)
                 {
-                    throw new InvalidOperationException($"Roslyn package version '{package.Version}' inconsistent with build version '{roslynBuildVersion}'");
+                    throw new InvalidOperationException($"Roslyn package version '{package.Version}' inconsistent with build version '{buildVersion}'");
```
Comment (jasonmalinowski) [inline src/RoslynInsertionTool/RoslynInsertionTool/RoslynInsertionTool.Packages.cs]: Does the "Roslyn" in there need to be dropped too?

File: src/RoslynInsertionTool/RoslynInsertionTool/RoslynInsertionTool.UpdateFiles.cs [hunk 2]
```diff
@@ -15,26 +15,40 @@ internal static string GetDevDivInsertionFilePath(BuildVersion version, string r
...
+
+            // For example: "\\cpvsbuild\drops\Roslyn\Roslyn-Project-System\DotNet-Project-System\20180111.1\packages"
+            var packagesPath = Path.Combine(GetBuildDirectory(version), "packages");
+            if (File.Exists(packagesPath))
+            {
+                return packagesPath;
+            }
+
+            Log.Error($"Unable to find packages path,  tried '{devDivPackagesPath}' and '{packagesPath}'");
+            throw new Exception("Unable to find packages directory");
```
Comment (jasonmalinowski) [inline src/RoslynInsertionTool/RoslynInsertionTool/RoslynInsertionTool.UpdateFiles.cs]: Throw same string as your Log.Error?

File: src/RoslynInsertionTool/RoslynInsertionTool/RoslynInsertionTool.Packages.cs (hunk 1 above)
Comment (jmarolf) [reply src/RoslynInsertionTool/RoslynInsertionTool/RoslynInsertionTool.Packages.cs]: its not the "Roslyn" version.  It the project system/LUT etc build version.

Comment (jasonmalinowski) [reply src/RoslynInsertionTool/RoslynInsertionTool/RoslynInsertionTool.Packages.cs]: Exactly, shouldn't the word "Roslyn" be removed then?

**Comment by jasonmalinowski** [comment]
(I guess I approve modulo the naming complaint in #136).

File: src/RoslynInsertionTool/RoslynInsertionTool/RoslynInsertionTool.Packages.cs (hunk 1 above)
Comment (jmarolf) [reply src/RoslynInsertionTool/RoslynInsertionTool/RoslynInsertionTool.Packages.cs]: Ah yes. sorry, thought you were talking about the variable change from `roslynBuildVersion` -> `buildVersion` not the message text.

**Comment by jmarolf** [comment]
moved naming changes over to https://github.com/dotnet/roslyn-tools/pull/136

**Pull request merged**